InterfacePython/TasmanianSG.in.py

```diff
@@ -644,6 +645,74 @@ def makeWaveletGrid(self, iDimension, iOutputs, iDepth, iOrder=1, liLevelLimits=
+                iDepth depends on sType
+                Example 1: sType == 'iptotal' will give a grid that
+                           interpolates exactly all polynomials of
+                           degree up to and including iDepth
+                Example 2: sType == 'qptotal' will give a grid that
+                           integrates exactly all polynomials of degree
+                           up to and including iDepth
+
+        sType: string identifying the tensor selection strategy
+              'level'     'curved'     'hyperbolic'     'tensor'
+              'iptotal'   'ipcurved'   'iphyperbolic'   'iptensor'
+              'qptotal'   'qpcurved'   'qphyperbolic'   'qptensor'
+
+        liAnisotropicWeights: list or numpy.ndarray of weights
+                              length must be iDimension or 2*iDimension
+                              the first iDimension weights
+                                                       must be positive
+                              see the manual for details
+
+        '''
+        if (iDimension <= 0):
+            raise TasmanianInputError("iDimension", "ERROR: dimension should be a positive integer")
+        if (iOutputs < 0):
+            raise TasmanianInputError("iOutputs", "ERROR: outputs should be a non-negative integer")
+        if (iDepth < 0):
+            raise TasmanianInputError("iDepth", "ERROR: depth should be a non-negative integer")
+        if (sType not in lsTsgGlobalTypes):
+            raise TasmanianInputError("sType", "ERROR: invalid type, see TasmanianSG.lsTsgGlobalTypes for list of accepted types")
+        pAnisoWeights = None
+        if (len(liAnisotropicWeights) > 0):
+            if (sType in lsTsgCurvedTypes):
+                iNumWeights = 2*iDimension
+            else:
+                iNumWeights = iDimension
+            if (len(liAnisotropicWeights) != iNumWeights):
+                raise TasmanianInputError("liAnisotropicWeights", "ERROR: wrong number of liAnisotropicWeights, sType '{0:s}' needs {1:1d} weights but len(liAnisotropicWeights) == {2:1d}".format(sType, iNumWeights, len(liAnisotropicWeights)))
+            else:
+                pAnisoWeights = (c_int*iNumWeights)()
+                for iI in range(iNumWeights):
+                    pAnisoWeights[iI] = liAnisotropicWeights[iI]
+
+        pLevelLimits = None
+        if (len(liLevelLimits) > 0):
+            if (len(liLevelLimits) != iDimension):
+                raise TasmanianInputError("liLevelLimits", "ERROR: invalid number of level limits, must be equal to iDimension")
+            pLevelLimits = (c_int*iDimension)()
+            for iI in range(iDimension):
+                pLevelLimits[iI] = liLevelLimits[iI]
+
+        if (sys.version_info.major == 3):
+            sType = bytes(sType, encoding='utf8')
+
+        self.pLibTSG.tsgMakeFourierGrid(self.pGrid, iDimension, iOutputs, iDepth, c_char_p(sType), pAnisoWeights, pLevelLimits)
```

Comment:
* you should test one call to the grid to see if it returns the correct set of points. Use pen and paper to write down the points (in Tasmanian order) for a small grid (2D with 10 - 20 points). Then, inside `testsTSG.in.py` add a test inside `def testFullCoverageA()` that makes a grid, reads the points and compares.